### ISSUE TITLE
Allow lower level access to Audio/Sound settings. ( SoundHandle.SetParameter )

### ISSUE THREAD

**Issue by DaddyFrosty** [Facepunch/sbox-issues#3431]
### What it is?

A while ago you were able to set sound parameters using a `SoundHandle`.

The following snippet allowed you to overwrite the sound max distance.
```cs
var soundHandle = Audio.Play( "item.m4.shoot");
soundHandle.SetParameter( "distance_min", 100f );
soundHandle.SetParameter( "distance_max", 5000f );
```

This was especially useful since the "SoundEvent" GameResource doesn't actually let you set a minimum distance, and frankly I'm not a fan of the "decibel" shit.
Like why `public float MinimumDistance => this.MaximumDistance * 0.05f;`?

### What should it be?

```cs
internal void SoundHandle.SetParameter(string name, float value);
internal void SoundHandle.SetParameter(string name, Vector3 value);
```
Should be changed back to public as they were. When it comes to audio there is practically zero control unless you wanna make your own audio engine...

**Comment by garrynewman** [comment]
IMO it would be stupid for us to expose SetParameter, purely because in the future those parameters might disappear or be renamed. In all likelyhood the sound system is going to end up getting much lower level customization, so I'm holding off until we get around to that.

Adding these common parameters are properties makes the most sense, since they're easy to convert to any potential future system.

**Comment by DaddyFrosty** [comment]
That's a very valid point, but I mean as long as you are using wont a lot of these parameters be the same? A lot of the parameter names I found were through just opening s&box in IDA, but in Googling them I found them in a string dump repo of Dota 2.

I do think that simpllifying it for the masses could be good, but some granular could also be handy.

But adding could be a great start if we're not getting SetParameter.
PS: Is there gonna be a way to get rid of doppler effects for certain entities / sounds.


```cs
public static SoundHandle SetMinDistance( this SoundHandle sndH, float minDistance )
{
	sndH.SetParameter( "distance_min", minDistance );
	return sndH;
}

public static SoundHandle SetMaxDistance( this SoundHandle sndH, float maxDistance )
{
	sndH.SetParameter( "distance_max", maxDistance );
	return sndH;
}
```
